content/frontend/angular-apollo/2-queries-loading-links.md

```diff
@@ -0,0 +1,322 @@
+
+![](https://i.imgur.com/R7YcAlj.png)
+
+
+### Writing the GraphQL Query
+
+You'll now load the actual link-items that are stored on the server. The first thing you need to do for that is define the GraphQL query that you want to send to the API.
+
+Here is what it looks like:
+
+```graphql
+query AllLinks {
+  allLinks {
+    id
+    createdAt
+    description
+    url
+  }
+}
+```
+
+You could now simply execute this query in a Playground and retrieve the results from your GraphQL server. But how can you use it inside your JavaScript code?
+
+
+### Queries with Apollo Client
+
+When using Angular with `apollo-angular` the `Apollo` service makes it easy to fetch GraphQL data.
+
+With this approach, all you need to do when it comes to data fetching is write the GraphQL query, then injecting the `Apollo`  service and use [`query`](http://dev.apollodata.com/angular2/queries.html) or [`watchQuery`](http://dev.apollodata.com/angular2/queries.html) method on the service directly that will fetch the data for you and finally make it available in your component.
+
+In general, the process for you to add some data fetching logic will be very similar every time:
+
+1. Write the query as a typescript constant using the `gql` parser function
+2. Initialize the property in your component
+2. Use the `Apollo` service to fetch the results of your `graphql` query using Observable
+3. Access the query results in the observable subscribe and assign it to the property in your component
+
+You will be writing your queries and mutations in a `graphql.ts` file and simply importing these queries and mutations into components as needed.
+
+<Instruction>
+
+Create a file within `src/app` called `graphql.ts`. This is where all of your queries and mutations will live.
+
+</Instruction>
+
+<Instruction>
+
+Open up `src/constants/graphql.js` and add your first query:
+
+```ts{2-3,5-15,17-21}(path=".../hackernews-vue-apollo/src/app/graphql.ts")
+import {Link} from './types';
+// 1
+import gql from 'graphql-tag'
+
+// 2
+export const ALL_LINKS_QUERY = gql`
+  query AllLinksQuery {
+    allLinks {
+      id
+      createdAt
+      url
+      description
+    }
+  }
+`;
+
+// 3
+export interface AllLinkQueryResponse {
+  allLinks: Link[];
+  loading: boolean;
+}
+
+```
+
+</Instruction>
+
+What's going on here?
+
+1. First, you import `gql` from the `graphql-tag` package. The `gql` function is used to parse the plain GraphQL code.
+2. Now you define the plain GraphQL query. The name `AllLinksQuery` is the _operation name_ and will be used by Apollo to refer to this query in its internals. You export this parsed query as `ALL_LINKS_QUERY` so you can easily import it into components.
+3. You declare also the response interface of `AllLinksQuery`
+
+Next, you will inject the `Apollo` service to the `LinkList` component and call this newly created query to fetch data.
+
+<Instruction>
```

Comment:
same problem as before, need to make multiple instruction blocks out of this one